### PR TITLE
Add docs to TabController

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Add docs to tab controller components

## Changelog
Add docs to tab controller components

### CODE REVIEW

File: src/components/tabController/apis/TabBar.api.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "TabController.TabBar",
+  "category": "layoutsAndTemplates",
+  "description": "TabController's TabBar component",
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "images": [],
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/TabBar.api.json:6]: We decided to remove empty keys

File: src/components/tabController/apis/pageCarousel.api.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "TabController.PageCarousel",
+  "category": "layoutsAndTemplates",
+  "description": "TabController's PageCarousel component",
+  "note": "You must pass `asCarousel` flag to TabController and render your TabPages inside a PageCarousel",
+  "extends": ["ScrollView"],
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "images": [],
+  "props": []
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/pageCarousel.api.json:9]: Remove empty keys

File: src/components/tabController/apis/tabBarItem.api.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "TabController.TabBarItem",
+  "category": "layoutsAndTemplates",
+  "description": "TabController's TabBarItem component",
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "images": [],
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/tabBarItem.api.json:6]: Also here

File: src/components/tabController/apis/tabController.api.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "TabController",
+  "category": "layoutsAndTemplates",
+  "description": "A performant solution for a tab controller with lazy load mechanism",
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "note": "This component is based on react-native-gesture-handler.\nOn Android, when using react-native-navigation, make sure to wrap your screen with gestureHandlerRootHOC",
+  "images": [],
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/tabController.api.json:7]: here

File: src/components/tabController/apis/tabController.api.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "TabController",
+  "category": "layoutsAndTemplates",
+  "description": "A performant solution for a tab controller with lazy load mechanism",
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "note": "This component is based on react-native-gesture-handler.\nOn Android, when using react-native-navigation, make sure to wrap your screen with gestureHandlerRootHOC",
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/tabController.api.json:6]: Shouldn't it be "notes" (plural)?

File: src/components/tabController/apis/TabBar.api.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "TabController.TabBar",
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/TabBar.api.json:2]: The name of the **file** should be with lowercase

File: src/components/tabController/apis/tabPage.api.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "TabController.TabPage",
+  "category": "layoutsAndTemplates",
+  "description": "TabController's TabPage component",
+  "example": "https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/TabControllerScreen/index.tsx",
+  "images": [],
```
Comment (Inbal-Tish) [inline src/components/tabController/apis/tabPage.api.json:6]: here

**Comment by Inbal-Tish** [comment]
@ethanshar What about prop's default values? I don't see any...

File: src/components/tabController/apis/TabBar.api.json (hunk 1 above)
Comment (ethanshar) [reply src/components/tabController/apis/TabBar.api.json:6]: I kept the images so it'll remind me to add ones.. cause technically we want all components to have images/gifs

File: src/components/tabController/apis/pageCarousel.api.json (hunk 2 above)
Comment (ethanshar) [reply src/components/tabController/apis/pageCarousel.api.json]: Removed only the props.. images kept same reason as previous comment

File: src/components/tabController/apis/tabController.api.json (hunk 5 above)
Comment (ethanshar) [reply src/components/tabController/apis/tabController.api.json:6]: You said yesterday we're using `note` cause we're noting the user 
It's just a big note (:

File: src/components/tabController/apis/TabBar.api.json (hunk 6 above)
Comment (ethanshar) [reply src/components/tabController/apis/TabBar.api.json]: fixed

**Comment by ethanshar** [comment]
> @ethanshar What about prop's default values? I don't see any...

Add some missing default values to props
And for the images key, as I mentioned in one of the comments, Im keeping it so Ill remember to add images to this component.

File: src/components/tabController/apis/tabController.api.json (hunk 5 above)
Comment (Inbal-Tish) [reply src/components/tabController/apis/tabController.api.json:6]: I thought on component-level it's "notes" and on prop-level it's "note". But ok. I'll change it on my side as well

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**